### PR TITLE
EVG-12387: fix linting and various style issues

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-12387

* Add doc comments for many things that did not have it.
* Enforce doc comments so that it's automated and I don't have to comb through code reviews for it.
* Move code around to proper locations and fix things that go against jasper coding style.

### CODE REVIEW

File: internal/executor/executor.go [hunk 1]
```diff
@@ -73,26 +73,37 @@ func (s Status) String() string {
 	}
 }
 
+// Before returns whether or not the given state occurs before the current
+// state.
 func (s Status) Before(other Status) bool {
 	return s < other
 }
 
+// BeforeInclusive is the same as Before but returns true if the given state is
```
Comment (julianedwards) [inline internal/executor/executor.go:82]: *but also returns true

File: internal/executor/executor.go [hunk 2]
```diff
@@ -73,26 +73,37 @@ func (s Status) String() string {
...
 func (s Status) BeforeInclusive(other Status) bool {
 	return s <= other
 }
 
+// After returns whether or not the given state occurs before the current state.
 func (s Status) After(other Status) bool {
 	return s > other
 }
 
+// AfterInclusive is the same as After but returns true if the given state is
```
Comment (julianedwards) [inline internal/executor/executor.go:93]: *but also returns true

File: internal/executor/executor.go [hunk 3]
```diff
@@ -73,26 +73,37 @@ func (s Status) String() string {
...
 	return s >= other
 }
 
+// Between returns whether or not the state occurs after the given lower state
+// and after the given upper state.
 func (s Status) Between(lower Status, upper Status) bool {
 	return lower < s && s < upper
 }
 
+// BetweenInclusive is the same as Between but returns true if the given
```
Comment (julianedwards) [inline internal/executor/executor.go:105]: *but also returns true

File: internal/executor/executor.go [hunk 4]
```diff
@@ -73,26 +73,37 @@ func (s Status) String() string {
...
+// state is identical ot the given lower or upper state.
```
Comment (julianedwards) [inline internal/executor/executor.go:106]: ot -> to

File: options/logger.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ const (
 // documentation for grip/send for more information on the various LogFormats.
 type LogFormat string
 
+// Constants represents formats for logging.
```
Comment (julianedwards) [inline options/logger.go:23]: representing

File: options/logger_producers.go [hunk 6]
```diff
@@ -179,20 +198,22 @@ func (opts *InMemoryLoggerOptions) Configure() (send.Sender, error) {
 // Sumo Logic Logger
 ///////////////////////////////////////////////////////////////////////////////
 
-// LogSumoLogic is the type name for the sumo logic logger.
+// LogSumoLogic is the name for the sumo logic logger.
```
Comment (julianedwards) [inline options/logger_producers.go:201]: Sumo Logic

File: scripting/testing.go [hunk 7]
```diff
@@ -23,7 +24,7 @@ type TestOptions struct {
 	Count   int           `bson:"count" json:"count" yaml:"count"`
 }
 
-// TestResults capture the data about a specific test run.
+// TestResult capture the data about a specific test run.
```
Comment (julianedwards) [inline scripting/testing.go:27]: captures

File: evergreen.yaml [hunk 8]
```diff
@@ -16,6 +16,11 @@ variables:
     commands:
       - func: run-make
         vars: { target: "${task_name}" }
+        command: subprocess.exec
```
Comment (julianedwards) [inline evergreen.yaml:19]: Do we want this?

Comment (Kimchelly) [reply evergreen.yaml]: No, I removed it.

**Pull request merged**